snuba/query/formatters/anonymize.py

```diff
@@ -0,0 +1,177 @@
+    formatter = expression_formatter_type()
+
+    return [
+        v
+        for v in [
+            _format_select(query, formatter),
+            _format_groupby(query, formatter),
+            _format_orderby(query, formatter),
+            _build_optional_string_node("ARRAY JOIN", query.get_arrayjoin(), formatter),
+            _build_optional_string_node("WHERE", query.get_condition(), formatter),
+            _build_optional_string_node("HAVING", query.get_having(), formatter),
+            _format_limitby(query, formatter),
+            _format_limit(query, formatter),
+            _format_offset(query, formatter),
+            PaddingNode(
+                "FROM",
+                StringQueryFormatter(expression_formatter_type).visit(
+                    query.get_from_clause()
+                ),
+            ),
+        ]
+        if v is not None
+    ]
+
+
+def _format_select(
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
+) -> StringNode:
+    selected_cols = [
+        e.expression.accept(formatter) for e in query.get_selected_columns()
+    ]
+    return StringNode(f"SELECT {', '.join(selected_cols)}")
+
+
+def _format_groupby(
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
+) -> Optional[StringNode]:
+    ast_groupby = query.get_groupby()
+    if ast_groupby:
+        selected_cols = [e.accept(formatter) for e in ast_groupby]
+        return StringNode(f"GROUPBY {', '.join(selected_cols)}")
+    return None
+
+
+def _format_orderby(
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
+) -> Optional[StringNode]:
+    ast_orderby = query.get_orderby()
+    if ast_orderby:
+        orderby = [
+            f"{e.expression.accept(formatter)} {e.direction.value}" for e in ast_orderby
+        ]
+        return StringNode(f"ORDER BY {', '.join(orderby)}")
+    else:
+        return None
+
+
+def _build_optional_string_node(
+    name: str, expression: Optional[Expression], formatter: ExpressionVisitor[str],
+) -> Optional[StringNode]:
+    return (
+        StringNode(f"{name} {expression.accept(formatter)}")
+        if expression is not None
+        else None
+    )
+
+
+def _format_limitby(
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
+) -> Optional[StringNode]:
+    ast_limitby = query.get_limitby()
+
+    if ast_limitby is not None:
+        return StringNode(
+            "LIMIT {} BY {}".format(
+                ast_limitby.limit, ast_limitby.expression.accept(formatter)
+            )
+        )
+
+    return None
+
+
+def _format_limit(
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
+) -> Optional[StringNode]:
+    ast_limit = query.get_limit()
+    return StringNode(f"LIMIT {ast_limit}") if ast_limit is not None else None
+
+
+def _format_offset(
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
+) -> Optional[StringNode]:
+    offset = query.get_offset()
+    return StringNode(f"OFFSET {query.get_offset()}") if offset is not None else None
```

Comment:
I think you should be able to reuse the methods we have in clickhouse/formatter/query.py as they are. They would also be well tested.